packages/api/src/v2/resourcePlugin.ts

```diff
@@ -138,10 +139,10 @@ export interface ResourcePlugin {
   configureResource?: (
     ctx: Context,
     inputs: Readonly<ProvisionInputs>,
-    provisionOutput: Readonly<ProvisionOutput>,
-    provisionOutputOfOtherPlugins: Readonly<Record<PluginName, ProvisionOutput>>,
+    envConfig: EnvConfig,
+    envProfile: EnvProfile,
     tokenProvider: TokenProvider
-  ) => Promise<Result<ProvisionOutput, FxError>>;
+  ) => Promise<Result<Void, FxError>>;
```

Comment:
Returning Void means plugins are expected to mutate envConfig and envProfile. Consider returning a new copy instead.

Reply:
If the API returns FxError, then core still need to persist the envProfile object, so just modify the input object makes the behavior more consistent.